File: backend/ServiceCatalog/Models/ServiceCatalog.cs
namespace ItServiceManagement.API.ServiceCatalog.Models;

public class ServiceCatalog
{
    public int Id { get; set; }

    public string ServiceName { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public string Category { get; set; } = string.Empty;

    public string RequestType { get; set; } = string.Empty;

    public string? ServiceOwner { get; set; }

    public string? EstimatedDeliveryTime { get; set; }

    public string DefaultPriority { get; set; } = string.Empty;

    public string SLA { get; set; } = string.Empty;

    public bool IsActive { get; set; } = true;
}

File: frontend/src/app/catalog/services/service-catalog.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface ServiceCatalogItem {
  id?: number;
  serviceName: string;
  description: string;
  category: string;
  requestType: string;
  serviceOwner: string;
  estimatedDeliveryTime: string;
  defaultPriority: string;
  sla: string;
  isActive: boolean;
}

@Injectable({ providedIn: 'root' })
export class ServiceCatalogService {
  private apiUrl = 'http://localhost:5103/api/servicecatalog';
  constructor(private http: HttpClient) {}
  getServices(): Observable<ServiceCatalogItem[]> { return this.http.get<ServiceCatalogItem[]>(this.apiUrl); }
  getService(id: number): Observable<ServiceCatalogItem> { return this.http.get<ServiceCatalogItem>(`${this.apiUrl}/${id}`); }
  createService(service: ServiceCatalogItem): Observable<ServiceCatalogItem> { return this.http.post<ServiceCatalogItem>(this.apiUrl, service); }
  updateService(id: number, service: ServiceCatalogItem): Observable<ServiceCatalogItem> { return this.http.put<ServiceCatalogItem>(`${this.apiUrl}/${id}`, service); }
  deleteService(id: number): Observable<void> { return this.http.delete<void>(`${this.apiUrl}/${id}`); }
}
